File: bessel.f90
real(8) function bessel(n, x)
  implicit none
  integer(4) i, n, m
  real(8) sum, x
  real(8), external :: fact
  sum=0.0d0
  do m=0, 20
    sum=sum+(-1)**m*(x/2.0d0)**(2*m+n)/(fact(m)*fact(m+n))
  end do
  bessel=sum
end function bessel

File: C/bessel.c
double fact(int);
double pow(double, double);
double bessel(int n, double x) {
  int i;
  double sum;
  sum=0.0;
  for(i=0; i<=15; i++) {
    sum+= pow(-1.0, (double)(i+n))*pow(x/2.0, (double)n+2.0*i)/
          (fact(i)*fact(i+n));
  }
  return sum;
}
